awx/main/tasks/system.py

```diff
@@ -597,11 +596,127 @@ def inspect_execution_and_hop_nodes(instance_list):
+    if active_task_ids is None:
+        logger.warning("No active task IDs retrieved from dispatcherd, skipping reaper")
+        return  # Failed to get task IDs, don't attempt reaping
+
+    # Run local reaper using tasks from dispatcherd
+    ref_time = datetime.now()  # No dispatch_time in dispatcherd version
+    logger.debug(f"Running reaper with {len(active_task_ids)} excluded UUIDs")
+    reaper.reap(instance=this_inst, excluded_uuids=active_task_ids, ref_time=ref_time)
+    # Always reap waiting tasks in the dispatcherd implementation
+    reaper.reap_waiting(instance=this_inst, excluded_uuids=active_task_ids, ref_time=ref_time)
+
+
+def _get_active_task_ids_from_dispatcherd():
+    """
+    Retrieve active task IDs from the dispatcherd control API.
+
+    Returns:
+        list: List of active task UUIDs
+        None: If there was an error retrieving the data
+    """
+    active_task_ids = []
+    try:
+        from dispatcherd.factories import get_control_from_settings
+
+        logger.debug("Querying dispatcherd API for running tasks")
+        ctl = get_control_from_settings()
+        running_data = ctl.control_with_reply('running')
+
+        # Extract UUIDs from the running data
+        if running_data and len(running_data) > 0:
+            logger.debug(f"Received {len(running_data)} response(s) from dispatcherd")
+            # Process running data: first item is a dict with node_id and task entries
+            data = running_data[0].copy()
+            data.pop('node_id', None)
+
+            # Extract task UUIDs from data structure
+            for task_key, task_value in data.items():
+                if isinstance(task_value, dict) and 'uuid' in task_value:
+                    active_task_ids.append(task_value['uuid'])
+                    logger.debug(f"Found active task with UUID: {task_value['uuid']}")
+                elif isinstance(task_key, str):
+                    # Handle case where UUID might be the key
+                    active_task_ids.append(task_key)
+                    logger.debug(f"Found active task with key: {task_key}")
+
+        logger.debug(f"Retrieved {len(active_task_ids)} active task IDs from dispatcherd")
+        return active_task_ids
+    except Exception:
+        logger.exception("Failed to get running tasks from dispatcherd")
+        return None
+
+
+# NOTE: This approach of registering alternative dispatcher implementations is a targeted solution
+# for specific functions (currently only cluster_node_heartbeat) and is not a general
+# solution for all tasks.
+try:
+    # Import and use the registry from publish module
+    from awx.main.dispatch.publish import ALTERNATIVE_TASK_IMPLEMENTATIONS
+
+    # Register the alternative implementation
+    ALTERNATIVE_TASK_IMPLEMENTATIONS[cluster_node_heartbeat.name] = adispatch_cluster_node_heartbeat
+    logger.info(f"Successfully registered dispatcherd method for {cluster_node_heartbeat.name}")
```

Comment:
This log gets noisy due to (what I think is) uWSGI cycling workers. It's fine for now. As we get closer, we might remove it after it's tested.